alacritty/src/input.rs

```diff
@@ -870,32 +891,61 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
             MouseState::Text
         }
     }
+
+    /// Handle automatic scrolling when selecting above/below the window.
+    fn update_selection_scrolling(&mut self, mouse_y: i32) {
+        let size_info = self.ctx.size_info();
+        let scheduler = self.ctx.scheduler();
+
+        // Scale constants by DPI.
+        let min_height = (MIN_SELECTION_SCROLLING_HEIGHT * size_info.dpr) as i32;
+        let step = (SELECTION_SCROLLING_STEP * size_info.dpr) as i32;
+
+        // Compute the height of the scrolling areas.
+        let end_top = max(min_height, size_info.padding_y as i32);
+        let height_bottom = max(min_height, size_info.padding_bottom() as i32);
+        let start_bottom = size_info.height as i32 - height_bottom;
+
+        // Get distance from closest window boundary.
+        let delta = if mouse_y < end_top {
+            end_top - mouse_y + step
+        } else if mouse_y >= start_bottom {
+            start_bottom - mouse_y - step
+        } else {
+            scheduler.unschedule(TimerId::SelectionScrolling);
+            return;
+        };
+
+        // Scale number of lines scrolled based on distance to boundary.
+        let delta = delta as isize / SELECTION_SCROLLING_STEP as isize;
```

Comment:
isn't it `delta / step` or it's intentional that way? If it's intentional, leave a comment why.

Reply:
Correct, thanks.